llvm/test/Transforms/LowerTypeTests/cfi-annotation.ll

```diff
@@ -0,0 +1,46 @@
+; RUN: opt -passes=lowertypetests %s -o %t.o
```

Comment:
After your last update, the test will also pass without your changes. I believe the issue is that you dropped the `!2 = !{i32 4, !"CFI Canonical Jump Tables", i32 0}` metadata, which is necessary.

Reply:
The issue turns out to be that I dropped attribute and type specifications:

- With the following the problem will reproduce:

` define i32 @bar(i32 noundef %0) #0 !type !8 !type !9 {`

- And with the following the problem doesn't reproduce:

` define i32 @bar(i32 noundef %0) {`

`#0`, `!8` and `!9` are defined as:

```
attributes #0 = { noinline nounwind optnone uwtable "frame-pointer"="non-leaf" "no-trapping-math"="true" "stack-protector-buffer-size"="8" "target-cpu"="generic" "target-features"="+fp-armv8,+neon,+v8a,-fmv" }
 60 !8 = !{i64 0, !"_ZTSFiiE"}
 61 !9 = !{i64 0, !"_ZTSFiiE.generalized"}
```

Also it appears that the test code can't be made too simple because otherwise indirect call target will be easy to figure out by optimizations and then CFI has nowhere to apply.